sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -265,14 +265,21 @@ abstract class Expression extends TreeNode[Expression] {
     case single => single :: Nil
   }
 
+  protected def flatArgumentStrings: Iterator[String] = flatArguments.map {
+    case e: Expression => e.argumentString
+    case arg => s"$arg"
```

Comment:
nit: `s"$arg"` -> `arg.toString` along with `e.argumentString`?

Reply:
Actually I tried `arg.toString` first, but there is one testcase got `NullPointerException`. Thus I use `s"$arg"` to avoid the `null` case.  Do we have other proper ways to solve this? Thanks!

Reply:
Ah, ok. Its ok to keep this as it is.